includes/plugins.php

```diff
@@ -118,17 +126,17 @@ function plugin_meta( $plugin_meta, $plugin_file, $plugin_data, $status ) {
  * @return void
  */
 function plugin_deactivation_warning() {
-	$message = __( 'Warning: This plugin provides additional enterprise-grade protective measures such as REST API security and disabling file editing in the dashboard.\n\nAre you sure you want to deactivate?', 'tenup' );
-?>
+	$message = esc_html__( 'Warning: This plugin provides additional enterprise-grade protective measures such as REST API security and disabling file editing in the dashboard.\n\nAre you sure you want to deactivate?', 'tenup' );
+	?>
 <script type="text/javascript">
 jQuery( document ).ready( function( $ ) {
 	$( '.wp-list-table.plugins tr[data-slug="10up-experience"] .deactivate' ).on( 'click', function( e ) {
-		if ( ! window.confirm( '<?php esc_html_e( $message ); ?>' ) ) {
+		if ( ! window.confirm( '<?php echo esc_js( $message ); ?>' ) ) {
```

Comment:
`esc_js()` is really meant for inline JS (like `onclick` attributes) and in modern practice probably doesn't need to be used at all. Why the change?

Reply:
I thought `esc_js` was right here. I could definitely be wrong. The description for the function is "Escape single quotes, htmlspecialchar ” &, and fix line endings.". Escaping single quotes is definitely what we are looking for. I'm pretty sure `window.confirm` doesn't support HTML. Either way `esc_html_e` is definitely not right.

Reply:
The `esc_js()` function itself does some pretty generic stuff, it's really about the intent of it I guess (and also that it's weirdly filterable). It does make sense for the single quotes thing though, I guess this isn't that far off from an inline action.